### PR TITLE
Enable nullability in ToolStripRenderer

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ToolStripRenderer

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6302)

### CODE REVIEW

**Comment by RussKie** [comment]
MC

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs [hunk 1]
```diff
@@ -36,10 +34,10 @@ public abstract class ToolStripRenderer
 
         private static readonly object EventRenderStatusStripSizingGrip = new object();
 
-        private static ColorMatrix disabledImageColorMatrix;
+        private static ColorMatrix? _disabledImageColorMatrix;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs:37]:
```suggestion
        private static ColorMatrix? s_disabledImageColorMatrix;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs [hunk 2]
```diff
@@ -628,12 +626,12 @@ protected virtual void OnRenderArrow(ToolStripArrowRenderEventArgs e)
...
                 //middle.X += (dropDownRect.Width % 2);
 
-                Point[] arrow = null;
+                Point[]? arrow = null;
 
                 // We need the Elvis operator here, since at design time at this point Item can be null.
                 if (e.Item?.DeviceDpi != previousDeviceDpi && DpiHelper.IsPerMonitorV2Awareness)
                 {
-                    previousDeviceDpi = e.Item.DeviceDpi;
+                    previousDeviceDpi = e.Item!.DeviceDpi;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs:634]: This looks like a bug introduced in https://github.com/dotnet/winforms/pull/639 - if `e.Item == null` we'll end up with a NRE here.
I think line:632 must look as
```cs
                if (e.Item is not null && e.Item.DeviceDpi != previousDeviceDpi && DpiHelper.IsPerMonitorV2Awareness)
```

/cc: @KlausLoeffelmann

Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs]: Sadly, yes,

**Review by RussKie — APPROVED** [review]

**Pull request merged**